modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -121,3 +121,72 @@ func (suite *KeeperTestSuite) TestChanUpgradeInit() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestRestoreChannel() {
+	var (
+		path *ibctesting.Path
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"succeeds when restore channel is set",
+			func() {},
+			true,
+		},
+		{
+			name: "fails when no restore channel is present",
+			malleate: func() {
+				// remove the restore channel
+				path.EndpointA.Chain.GetSimApp().IBCKeeper.ChannelKeeper.DeleteUpgradeRestoreChannel(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+			},
+			expPass: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			upgradeSequence := uint64(1)
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			suite.coordinator.Setup(path)
+
+			path.EndpointA.ChannelConfig.Version = fmt.Sprintf("%s-v2", mock.Version)
+
+			originalChannel := path.EndpointA.GetChannel()
+
+			err := path.EndpointA.ChanUpgradeInit(path.EndpointB.Chain.GetTimeoutHeight(), 0)
```

Comment:
updated the test to broadcast the message directly as the updated channel gets written in `ibckeeper.OnChanUpgradeInit` instead of `channelkeeper.OnChanUpgradeInit`.